homeassistant/components/sensor/zha.py

```diff
@@ -100,6 +103,35 @@ def attribute_updated(self, attribute, value):
         self._state = result.get(self.value_attribute, self._state)
 
 
+class BatterySensor(Sensor):
+    """ZHA generic battery sensor."""
+
+    value_attribute = 32
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return '%'
```

Comment:
Battery voltage is != "battery_percentage_remaining"

I'd use attrid = 33 as the value attributes which translates into "battery_percentage_remaining". You also would need to configure attribute reporting for attr_id=33. The problem is: Centralite sensors unlike SmartThings sensors do not report "battery_percentage_remaining" but this could be addressed through zigpy quirks, which I was actually working on. For now I have linear translation of 'battery_voltate' to 'battery_percentage_remaining', but was working on a reverse exponential translation like in #12515 just want to have a function with defined max/min battery voltage translating into % remaining

Reply:
Interesting. I realised that this wasnt battery percentage, I was thinking that a quirk could be used to convert this to percentage later on. Maybe I will check out the battery_percentage_remaining function and see how that goes.

Reply:
this is what I have for WIP https://github.com/Adminiuga/zigpy/commits/bat-percent-remain the only thing stopping me from submitting this is the linear battery_voltage --> battery_percentage_remaining conversion as I'd like it to be inverse exponential. Xiaomi battery voltage parsing is based on/inspired by Yoda-X code https://github.com/Yoda-x/ha-zha-new/blob/master/custom_components/device/lumi_sensor_wleak_aq1.py

Reply:
Looking in the zigbee.db file, I don't see a battery_percentage_remaining attribute for cluster one on my Smartthings sensors, but you mention that smarthing sensors do report this. Does this mean a change needs to happen in zigby?

Reply:
I guess that depends on the sensor. The one I've got https://www.amazon.com/Samsung-SmartThings-Multipurpose-Sensor-GP-U999SJVLAAA or https://www.amazon.com/Samsung-SmartThings-Water-Sensor-GP-U999SJVLCAA/dp/B07F951JDP do report "battery_percentage_remaining" see https://community.home-assistant.io/t/zha-zigbee-tested-devices-please-add-your-device-results/17718/431?u=quatuor and some probably not. 

Centralite Motion "Iris" sensors do not report "battery_percentage_remaining". The problem I see with reporting the battery % remaining based on battery voltage means that we either need to assume that all sensors have 3v batteries, which might not be always the case, like in https://www.sinopetech.com/en/boutique/all-products/smart-water-valve-3-4-in/ or create battery sensor based on model&manufacturer but in this case you maybe better going with "battery % remaining" attribute if the device supports it or implement it using zigpy quirks. Implementing it through zigpy quirks is pretty much equivalent to implementing battery sensor based on manufacturer/model, just moves abstraction layer to zigpy.

Reply:
Do you have the new smartthings sensors? I have the older variety of both the multipurpose and the water sensor.

Reply:
I agree it makes sense to use battery percentage if each sensor varies with the voltage, and use a quirk to implement it if its missing.

Reply:
Great, I'm going to try and create a smartthings quirk based on your example to write the percentage battery attribute.

Reply:
I implemented the quirk from your example above, but haven't had any luck getting a battery level reported, just get "Unknown".